model/modules/model/parse_country_model_module.py

```diff
@@ -71,6 +84,66 @@ def _apply_append_after(self, model: Model, yaml: Any) -> None:
           parent.children = (parent.children[:insert_idx + 1] + nodes +
                              parent.children[insert_idx + 1:])
 
+  def _find_lowest_common_ancestor(self, model: Model,
```

Comment:
I think the approach looks good. It is not clear to me that it fully works in case the sinthesized node contains only one children.

One (optional) alternative approach is the following:
1. Pick any children of the sinthesized node. 
2. Move up in the hierarchy using the parent links. 
3. Find the first parent that `is_ancestor` to all children. 

The approach requires a small helper function `is_ancestor(node_a, node_b)`, as opposed to the `find_path_to_node`. 

One advantage is that we can use the `is_ancestor` function to validate that none of the sinthesized nodes constituents are ancestors of each other (i.e. there is at most one constituent in all root to leaf paths).